app/actions/websocket/threads.ts

```diff
@@ -27,21 +29,27 @@ export function handleThreadReadChanged(msg: WebSocketMessage) {
             const thread = getThread(state, msg.data.thread_id);
 
             // Mark only following threads as read.
-            if (thread?.is_following) {
-                dispatch(
-                    handleReadChanged(
-                        msg.data.thread_id,
-                        msg.broadcast.team_id,
-                        msg.data.channel_id,
-                        {
-                            lastViewedAt: msg.data.timestamp,
-                            prevUnreadMentions: thread.unread_mentions,
-                            newUnreadMentions: msg.data.unread_mentions,
-                            prevUnreadReplies: thread.unread_replies,
-                            newUnreadReplies: msg.data.unread_replies,
-                        },
-                    ),
-                );
+            if (thread) {
+                const selectedPost = getSelectedPost(state);
+                if (selectedPost?.id !== thread.id) {
+                    dispatch(updateThreadLastViewedAt(thread.id, msg.data.timestamp));
```

Comment:
is there a way to batch this two actions ?

Reply:
They are of different type `GenericAction` and `AsyncFunc`. `batchActions` is throwing TS error when including `AyncFunc` .. Can I typecast to GenericAction?

Reply:
Not really, you would need to add some sort of parameter so that the async function does not dispatch to the reducers but instead return the action to be dispatched, then you could batch them.. there is code in the codebase that references, I can look it up for you if you want.. that said, will dispatching this actions as is cause a huge load running every mounted component mapStateToProps? If the answer is yes, then consider batching